lib/raxx/server.ex

```diff
@@ -183,14 +156,65 @@ defmodule Raxx.Server do
   """
   @callback handle_info(any(), state()) :: next
 
-  defmacro __using__(options) do
+  use Raxx.View, template: "server.html.eex", arguments: [:module]
+
+  defmacro __using__(_options) do
     quote do
       @behaviour unquote(__MODULE__)
+      import Raxx
 
-      use Raxx.NotFound, unquote(options)
+      @impl unquote(__MODULE__)
+      def handle_head(_request, _state) do
+        response(:not_found)
+        |> Raxx.Server.render(__MODULE__)
+      end
 
-      import Raxx
-      alias Raxx.{Request, Response}
+      @impl unquote(__MODULE__)
+      def handle_data(data, state) do
+        import Logger
+        Logger.warn("Received unexpected data: #{inspect(data)}")
+        {[], state}
+      end
+
+      @impl unquote(__MODULE__)
+      def handle_tail(trailers, state) do
+        import Logger
+        Logger.warn("Received unexpected trailers: #{inspect(trailers)}")
+        {[], state}
+      end
+
+      @impl unquote(__MODULE__)
+      def handle_info(message, state) do
+        import Logger
+        Logger.warn("Received unexpected message: #{inspect(message)}")
+        {[], state}
+      end
+
+      defoverridable unquote(__MODULE__)
+      @before_compile unquote(__MODULE__)
+    end
+  end
+
+  # DEBT Remove this for 1.0 release
+  defmacro __before_compile__(_env) do
+    quote do
+      try do
+        defoverridable handle_request: 2
+
+        if !Module.get_attribute(__MODULE__, :raxx_safe_server) do
```

Comment:
What's the idea behind `:raxx_safe_server` - is it leftover from a different approach, or are you planning to use it for something?

If there's some idea behind it, could you document it in a comment (it probably doesn't need to get to moduledocs)?

Reply:
It's a way to switch of the warning if you decide to use the streaming interface and write a handle_request for some reason